File: week6/sentimental-credit/credit.py
# TODO
from cs50 import get_float, get_int, get_string


x = get_int("Number: ")
count=1
a=0
b=0
type=0
type2=0
while(x is not 0):
    temp=x%10
    x=x//10
    if(x<100 and x>10):
        type=x
    if(x>0 and x<10):
        type2=x
    if((count%2)==0):
        temp=temp*2
        temp1=temp%10
        temp2=temp//10
        temp=temp1+temp2
        a=a+temp
    else:
        b=b+temp
    count+=1
count=count-1

if((a+b)%10==0):
    if((type==34 or type==37) and count==15):
        print("AMEX\n")
    elif((type==51 or type==52 or type==53 or type==54 or type==55)  and (count==16)) :
        print(" MASTERCARD\n")
    elif((count == 13 or count==16) and type2==4):
        print("VISA\n")
    else:
        print("INVALID\n")
else:
    print("INVALID\n")

File: week1/credit/credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    long x=0;
    int count=1;
    int a=0;
    int b=0;
    int type=0;
    int type2=0;
    x=get_long("Number:");

    while(x!=0){
        int temp=x%10;
        x=x/10;
        if(x<100&&x>10){
            type=x;
        }
        if(x>0&&x<10){
            type2=x;
        }
        if((count&1)==0){
            temp=temp*2;
            int temp1=temp%10;
            int temp2=temp/10;
            temp=temp1+temp2;
            a=a+temp;
        }else{
            b=b+temp;
        }
        count++;
    }

    count=count-1;


    if((a+b)%10==0){
        if((type==34 ||type==37) && count==15){
            printf("AMEX\n");
            return 0;
            }
        if((type==51 ||type==52||type==53||type==54||type==55)&&(count==16)) {
            printf(" MASTERCARD\n");
            return 0;
            }
        if((count == 13 || count==16)&&type2==4){
            printf("VISA\n");
            return 0;
        }
        printf("INVALID\n");
    }else{
        printf("INVALID\n");
    }
}
